blockindex/contractstaking/cache.go

```diff
@@ -350,7 +355,7 @@ func (s *contractStakingCache) getBucketInfoByCandidate(candidate address.Addres
 	m := make(map[uint64]*bucketInfo)
 	for k, v := range s.candidateBucketMap[candidate.String()] {
 		if v {
-			m[k] = s.bucketInfoMap[k]
+			m[k] = s.bucketInfoMap[k].clone()
```

Comment:
same here, we can remove this fun

it is only used in `func (s *contractStakingCache) BucketsByCandidate`, we can directly access `s.bucketInfoMap` in there

Reply:
fixed